fedimint-core/src/db/mod.rs

```diff
@@ -1293,6 +1298,15 @@ pub struct DatabaseTransaction<'tx, Cap = NonCommittable> {
     capability: marker::PhantomData<Cap>,
 }
 
+impl<'tx, Cap> fmt::Debug for DatabaseTransaction<'tx, Cap> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_fmt(format_args!(
+            "DatabaseTransaction {{ tx: {:?}, decoders={:?} }}",
+            self.tx, self.decoders
+        ))
+    }
+}
+
```

Comment:
nit: why not use [`std::fmt::DebugStruct`](https://doc.rust-lang.org/std/fmt/struct.DebugStruct.html)?

Reply:
TIL;

But looking at it - it doesn't look like that much of an improvement. Will use next time I'm doing anything like that. If I remember. :D

Reply:
I think I learned about it reviewing a Fedimint PR :laughing: probably by @maan2003, he always knows about these cool things!